source/include/VirtualInstrument.h

```diff
@@ -42,10 +44,31 @@ class VIClump;
 "    e(SubString ClumpSource)       \n" \
 "))"
 
-class EventInfo {
- public:
+struct EventStructInfo {
     OccurrenceCore eventOccurrence;
     Int32 setCount;
+    EventQueueID staticQID;
+    EventStructInfo() : eventOccurrence(), setCount(0), staticQID(0) { }
+};
+
+struct EventControlInfo {
+    EventOracleIndex eventOracleIndex;
+    RefNum refnum;
+    explicit EventControlInfo(EventOracleIndex eoIdx = 0, RefNum ref = 0) : eventOracleIndex(eoIdx), refnum(ref) { }
+};
+
+struct EventInfo {
+    typedef std::map<EventControlUID, EventControlInfo> ControlIDInfoMap;
```

Comment:
Hi Craig, Can we change EventControlUID from Integer to string? ControlRefInfo.ControlTag is already a string. Also, there are plans to change the ControlID in source model to be GUIDs / 64-bit integers.

Reply:
We talked about this in person.  Instead of resolving the controlRef to a controlID in the EventSpec storage, we'll leave it as a controlRef and match on that. The controlID/GUID string will stay in the RefNumStorage as a string, which it already is.
We will remove the controlID from the OccurEvent JS CEntryPoint, since it can be looked up from the eventOracleIdx and we don't want to have to pass a string.

This will happen in a follow-on PR and doesn't need to block this merge.